packages/react-core/src/components/Modal/ModalContent.tsx

```diff
@@ -138,7 +152,12 @@ export const ModalContent: React.FunctionComponent<ModalContentProps> = ({
     <ModalBox
       id={boxId}
       style={boxStyle}
-      className={className}
+      className={css(
+        className,
+        isModalAlertVariant(titleIconVariant) &&
+          modalStyles.modifiers[titleIconVariant as 'success' | 'warning' | 'info' | 'danger' | 'default'],
+        titleIconVariant && modalStyles.modifiers.icon
```

Comment:
`modalStyles.modifiers.icon` should go on `<ModalBoxTitle>` when there is an icon passed. Also when there is a modal, the value of the `title` prop should be wrapped in `<span class="pf-c-modal-box__title-text">`